apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/SpannerCommandTests.cs

```diff
@@ -760,6 +761,36 @@ public void PdmlRetriedOnEosError()
                 It.IsAny<CallSettings>()), Times.Exactly(3));
         }
 
+        [Fact]
+        public async Task ParallelMutationCommandsOnAmbientTransaction_OnlyCreateOneSpannerTransactionAsync()
+        {
+            Mock<SpannerClient> spannerClientMock = SpannerClientHelpers
+                .CreateMockClient(Logger.DefaultLogger, MockBehavior.Strict);
+            spannerClientMock
+                .SetupBatchCreateSessionsAsync()
+                .SetupBeginTransactionAsync()
+                .SetupCommitAsync();
+
+            using (TransactionScope scope = new TransactionScope(TransactionScopeAsyncFlowOption.Enabled))
+            {
+                SpannerConnection connection = BuildSpannerConnection(spannerClientMock);
+
+                var tables = new string[] { "TAB1", "TAB2", "TAB3" };
+                await Task.WhenAll(tables.Select(table =>
+                {
+                    using var cmd = connection.CreateInsertCommand(table);
+                    return cmd.ExecuteNonQueryAsync();
+                }));
+                scope.Complete();
+            }
+
+            spannerClientMock.Verify(client => client.CommitAsync(
```

Comment:
The verification in the next line superseeds this one. I think this one can be removed.

Reply:
I think it's better to have both, as they do check two different things:
1. The first verification checks that there is __only one__ `Commit` call, regardless of the number of mutations in the request.
2. The second verification checks that the only `Commit` call actually contains 3 mutations.

Removing the first verification would mean that if two commit requests were sent from the client, one without any mutations and the other with 3 mutations, the test would still pass. That is one of the scenarios that we want to verify is not happening. (Checking the number of `BeginTransaction` calls is not a good measure, as multiple transactions could be prepared by the session pool.)

Reply:
Ah yes, you are right.